src/core/server/base.js

```diff
@@ -112,6 +112,22 @@ function baseServer(routes, createStore, { appInstanceName = appName } = {}) {
       ) : {};
 
       const lang = getLangFromRouter(renderProps);
+
+      if (renderProps.params && renderProps.params.lang) {
+        const origLang = renderProps.params.lang;
+        if (lang !== origLang) {
+          // Remove the first slash and split on the /
+          const urlParts = req.originalUrl.replace(/^\//, '').split('/');
```

Comment:
I think you can destructure here, unless something weird happens because of node. Either way you shouldn't need to pop off that first slash, and increment the indexes by 1.

```js
const [ _, origLang, ...rest ] = url.split('/');
res.redirect(301, ['', lang, ...rest].join('/'))